Remotion/Data/DomainObjects/Persistence/Rdbms/Model/IRdbmsStructuredTypeDefinitionVisitor.cs

```diff
@@ -0,0 +1,29 @@
+// This file is part of the re-motion Core Framework (www.re-motion.org)
+// Copyright (c) rubicon IT GmbH, www.rubicon.eu
+//
+// The re-motion Core Framework is free software; you can redistribute it
+// and/or modify it under the terms of the GNU Lesser General Public License
+// as published by the Free Software Foundation; either version 2.1 of the
+// License, or (at your option) any later version.
+//
+// re-motion is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+//
+using Remotion.Data.DomainObjects.Persistence.Rdbms.Model;
+using Remotion.Data.DomainObjects.Persistence.Rdbms.SqlServer.Model;
+
+namespace Remotion.Data.DomainObjects.Persistence.Rdbms
+{
+  /// <summary>
+  /// Defines the API for the structured type definition visitor implementations.
+  /// </summary>
+  public interface IRdbmsStructuredTypeDefinitionVisitor
+  {
+    void VisitTableTypeDefinition (SqlTableTypeDefinition tableTypeDefinition);
```

Comment:
design issue: SqlTableTypeDefintiion is SqlServer-specific? Then it shouldn't be part of the rdbms-namespace but lower. I didn't search everywhere so I can't say if the name is incorrect or if the layering needs work. 

From a quick google search, it looks like Table-Type exists in PostgreSql nd Oracle, too. This means we should move the SqlTableTypeDefinition to the same level as TableDefiniton and drop the "Sql"